### PR TITLE
Update to Mixxx version 2.4 with updated dependencies

### PR DESCRIPTION
**Pull request by djantti**
Mixxx 2.4 is nearing release, so I'm already submitting this PR in hopes that it will be tested, reviewed and ready to go as close to final release day as possible. At the moment it's using the 2.4 git branch, but I'll update the manifest once the 2.4 tarball is available.

Changes include:

- Update to Mixxx 2.4 with new dependencies
- Slightly reorganized manifest file
- Use Wayland by default
- Existing dependencies updated to latest versions
- Everything checked for unused build options and features
- RelWithDebInfo build type for all CMake modules
- More unnecessary binaries and config files removed

I've tested my local build of this flatpak fairly extensively and used it on gigs without issues. Version 2.4 is a big update for Mixxx, so let me know if anything should be changed or improved. Happy DJing!

### CODE REVIEW

File: org.mixxx.Mixxx.yaml [hunk 1]
```diff
@@ -3,38 +3,53 @@ runtime: org.kde.Platform
 runtime-version: '5.15-23.08'
 sdk: org.kde.Sdk
 command: mixxx
+rename-icon: mixxx
+
+# Tag and mark as beta until 2.4 final release
+tags:
+  - beta
+desktop-file-name-suffix: ' (beta)'
+
```
Comment (hfiguiere) [inline org.mixxx.Mixxx.yaml:12]:
```suggestion
```

no. Because this is the stable branch.

File: org.mixxx.Mixxx.yaml [hunk 2]
```diff
@@ -3,38 +3,53 @@ runtime: org.kde.Platform
...
+  # Wayland is used by default
   - --socket=wayland
-  - --socket=x11
-  # JACK support via PipeWire
-  - --filesystem=xdg-run/pipewire-0
-  # PulseAudio isn't actually helpful for Mixxx, but this allows access to ALSA
-  # devices too since https://github.com/flatpak/flatpak/pull/3663
-  - --socket=pulseaudio
-  # Only used for MusicBrainz lookups
+  - --socket=fallback-x11
```
Comment (hfiguiere) [inline org.mixxx.Mixxx.yaml:19]: do you test the plugins?

File: org.mixxx.Mixxx.yaml [hunk 3]
```diff
@@ -3,38 +3,53 @@ runtime: org.kde.Platform
...
+  - --socket=pulseaudio
+
+  # Talk to UPower and RTKit services on the session bus
   - --system-talk-name=org.freedesktop.UPower
-  - --env=ALSA_CONFIG_PATH=
   - --system-talk-name=org.freedesktop.RealtimeKit1
 
-rename-icon: mixxx
+  # Define LV2 plugin location and ignore ALSA configuration
+  - --env=LV2_PATH=/app/extensions/Plugins/lv2
```
Comment (hfiguiere) [inline org.mixxx.Mixxx.yaml:37]:
```suggestion
  # gnore ALSA configuration
```

LV2_PATH is properly set in the wrapper script.

**Comment by hfiguiere** [comment]
Build fail because appstream file doesn't validate
```
        "style-invalid         : <li> is too long [Migration to Qt6 (work in progress) #4052 #4295 #4293 #4294 #4291 #4290 #4300 #4302 #4289 #4292 #4299 #4051 #4303 #4305 #4304 #4306 #4308 #4309 #4322 #4373 #4371 #4375 #4378 #4381 #4380 #4376 #4379 #4372 #4377 #4387 #4391 #4392 #4395 #4397 #4396 #4402 #4405 #4394 #4404 #4401 #4400 #4403 #4407 #4399 #4406 #4420 #4415 #4417 #4419 #4416 #4418 #4547 #4433 #4434 #4441 #4445 #4446 #4444 #4436 #4437 #4440 #4430 #4435 #4443 #4439 #4442 #4438 #4449 #4451 #4453 #4478 #4479 #4506 #4556 #4554 #4555 #4552 #4549 #4788 #4620 #4641 #4649 #4679 #10928 #10927 #11063 #11153 #11180 #11167 #11181 #11266 #11312 #11389 #4393 #12268 #12240 #12056 #12430 #12029] maximum is 500 chars",
        "style-invalid         : <li> is too long [Misc. refactorings #3154 #2870 #2872 #2978 #2969 #3016 #3320 #3356 #3453 #3487 #3558 #3685 #3741 #3744 #3753 #3761 #3834 #3842 #3853 #3874 #3883 #3922 #3947 #3974 #4024 #4026 #4034 #4038 #4039 #4043 #4053 #4072 #4097 #4096 #4118 #4130 #4129 #4109 #4106 #4131 #4140 #3032 #4110 #4173 #4178 #4194 #4197 #4190 #4212 #4223 #4238 #4236 #4320 #4325 #4203 #3861 #3514 #3274 #3182 #4343 #4358 #4388 #4427 #4341 #4473 #4464 #4481 #4527 #4534 #4537 #4539 #4541 #4543 #4546 #4542 #4559 #4574 #4581 #4582 #4598 #4596 #4592 #4784 #4604 #4603 #4611 #4605 #4583 #4636 #4644 #4653 #4655 #4609 #4647 #4660 #4663 #4661 #4670 #4672 #4674 #4618 #4690 #4724 #4730 #4731 #4732 #4728 #4737 #4739 #4745 #4765 #4767 #4797 #4782 #4800 #4791 #4848 #4856 #4860 #4867 #4888 #4908 #4903 #8818 #4907 #10897 #10964 #10963 #11048 #11149 #11169 #11318 #11331 #11338 #4651 #11336 #11282 #11421 #11431 #11434 #11301 #11688 #11685 #12000 #12003 #11684 #12031 #12008 #12054 #11768 #11893 #11636 #12026 #12230 #12375 #11619 #11917 #11936 #12033 #12263 #12212 #12066 #12065] maximum is 500 chars",
        "style-invalid         : <li> is too long [Note: Mixxx does not yet support multiple loops per track. We are working on this for Mixxx 2.4 . In Mixxx 2.3, if you import a track with multiple loops from Serato, Mixxx will use the first loop cue as the single loop Mixxx currently supports. The imported loops are still stored in Mixxx's database and are treated as hotcues in Mixxx 2.3. If you do not delete these hotcues, they will be usable as loops in Mixxx 2.4. Serato keeps loops and hotcues in separate lists, but Mixxx does not, so loops from Serato are imported starting as hotcue 9.] maximum is 500 chars",
        "tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]"
```

File: org.mixxx.Mixxx.yaml (hunk 2 above)
Comment (hfiguiere) [reply org.mixxx.Mixxx.yaml:19]: I mean their UI. On a Wayland session.

Comment (hfiguiere) [reply org.mixxx.Mixxx.yaml:19]: The question is rhetorical. It's just to show you how this breaks.

File: org.mixxx.Mixxx.yaml [hunk 4]
```diff
@@ -3,38 +3,53 @@ runtime: org.kde.Platform
...
+  # Used by X11 and MusicBrainz metadata lookups
+  - --share=ipc
   - --share=network
+
+  # Permit PipeWire and user music directory access
+  - --filesystem=xdg-run/pipewire-0
   - --filesystem=xdg-music
-  - --persist=.mixxx
-  - --device=all
-  - --metadata=X-DConf=migrate-path=/org/mixxx/Mixxx
```
Comment (hfiguiere) [inline org.mixxx.Mixxx.yaml:20]: why did you remove this?

File: org.mixxx.Mixxx.yaml [hunk 5]
```diff
@@ -3,38 +3,53 @@ runtime: org.kde.Platform
...
-  - --env=ALSA_CONFIG_PATH=
   - --system-talk-name=org.freedesktop.RealtimeKit1
 
-rename-icon: mixxx
+  # Define LV2 plugin location and ignore ALSA configuration
+  - --env=LV2_PATH=/app/extensions/Plugins/lv2
+  - --env=ALSA_CONFIG_PATH=
+
+  # App data storage in org.mixxx.Mixxx/.mixxx
+  - --persist=.mixxx
```
Comment (hfiguiere) [inline org.mixxx.Mixxx.yaml:41]: this should be fixed upstream.

File: org.mixxx.Mixxx.yaml (hunk 2 above)
Comment (hfiguiere) [reply org.mixxx.Mixxx.yaml:19]: Including the plugins UI?

File: mixxx-metainfo.patch [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+-   #11266
+-   #11312
+-   #11389
+-   #4393
+-   #12268
+-   #12240
+-   #12056
+-   #12430
+-   #12029
++   ...
```
Comment (hfiguiere) [inline mixxx-metainfo.patch:40]: this should just be a PLAIN ENGLISH high level description of the changes. Not bug numbers. This is meant to be read by **users**

**Comment by hfiguiere** [comment]
the problem is the wrapper script is missing the `export` statement. (see at bottom of the manifest)

**Comment by hfiguiere** [comment]
we probably did never notice because there was the `--env`.

File: mixxx-metainfo.patch (hunk 6 above)
Comment (hfiguiere) [reply mixxx-metainfo.patch:40]: This is the place for user visible change logs. It's whats displayed on flathub, on GNOME software or KDE discovery, and some developers even use to display it in their application.

File: org.mixxx.Mixxx.yaml (hunk 2 above)
Comment (Be-ing) [reply org.mixxx.Mixxx.yaml:14]: Mixxx does not support LV2 plugin GUIs, so the point is moot.

File: mixxx-metainfo.patch (hunk 6 above)
Comment (Be-ing) [reply mixxx-metainfo.patch:40]: This is autogenerated upstream from release announcements, which is why this patch is here to meet Flathub's requirements.

File: org.mixxx.Mixxx.yaml (hunk 5 above)
Comment (Be-ing) [reply org.mixxx.Mixxx.yaml:38]: Yes, and easier said than done. That won't be done for 2.4.

File: org.mixxx.Mixxx.yaml [hunk 7]
```diff
@@ -44,175 +56,290 @@ cleanup:
...
 
+  # Audio time stretching and pitch shifting library
   - name: rubberband
     buildsystem: meson
+    config-opts:
+      # Disable tests, command line tools and audio plugins
+      - -Dtests=disabled
+      - -Dcmdline=disabled
+      - -Dladspa=disabled
+      - -Dlv2=disabled
```
Comment (Be-ing) [inline org.mixxx.Mixxx.yaml:210]: Why disable the LV2 plugin? Mixxx could potentially use that, though there may be more work to get that working inside the flatpak.

File: org.mixxx.Mixxx.yaml [hunk 8]
```diff
@@ -44,175 +56,290 @@ cleanup:
...
+        sha256: 2bb837fe00932442ca90e185af8a468f7591df0c002b4a9e27a1bced1563ac84
 
+  # Audio processing Library
   - name: soundtouch
     buildsystem: cmake-ninja
+    config-opts:
+      # Disable soundstretch utility and Neon SIMD instructions
+      - -DSOUNDSTRETCH=OFF
+      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
+      - -DNEON=OFF
```
Comment (Be-ing) [inline org.mixxx.Mixxx.yaml:223]: Why disable NEON? There's an aarch64 build.

File: org.mixxx.Mixxx.yaml [hunk 9]
```diff
@@ -44,175 +56,290 @@ cleanup:
...
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
     sources:
       - type: archive
-        url: https://codeberg.org/tenacityteam/libid3tag/archive/0.16.2.tar.gz
-        sha256: 78b593395b03788cf9c6475b88ce0cbdf0476aaec5cf6f5339f04c2ffcae0704
+        url: https://codeberg.org/tenacityteam/libid3tag/archive/0.16.3.tar.gz
+        sha256: 3b65150454818e3e46d113891e0e0c335516315c51f80d87565f6124b892bff4
 
+  # Power management interface
```
Comment (Be-ing) [inline org.mixxx.Mixxx.yaml:239]:
```suggestion
  # Power management interface, used for battery level indicator
```

File: org.mixxx.Mixxx.yaml [hunk 10]
```diff
@@ -44,175 +56,290 @@ cleanup:
...
+    cleanup:
+      - /bin
+      - /etc
+      - /libexec
+      - /var
+      - /share/dbus-1
+      - /share/installed-tests
 
-  # Use fdk-aac fork without HE-AAC to avoid patent issues
-  # https://bugzilla.redhat.com/show_bug.cgi?id=1501522#c112
```
Comment (Be-ing) [inline org.mixxx.Mixxx.yaml:146]: keep the URL in the comment

File: org.mixxx.Mixxx.yaml [hunk 11]
```diff
@@ -44,175 +56,290 @@ cleanup:
...
-    config-opts:
-      - -DUSE_SYMLINKS=OFF
+    config-opts:
+      # Disable udev rules install
       - -DINSTALL_USER_UDEV_RULES=OFF
-    build-options:
-      cxxflags: "-Wno-error=restrict"
+      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
     post-install:
       - desktop-file-edit --set-key="Icon" --set-value=${FLATPAK_ID} ${FLATPAK_DEST}/share/applications/org.mixxx.Mixxx.desktop
```
Comment (hfiguiere) [inline org.mixxx.Mixxx.yaml:330]:
```suggestion
```

This is handled by `rename-icon` automatically. (I know it was already there)

File: org.mixxx.Mixxx.yaml (hunk 8 above)
Comment (Be-ing) [reply org.mixxx.Mixxx.yaml:223]: Interesting... here's what [SoundTouch's build system does](https://codeberg.org/soundtouch/soundtouch/src/commit/74514f559766cd24fc87aa278a5c8722c028b8ae/CMakeLists.txt#L62):

```cmake
if(CMAKE_SYSTEM_PROCESSOR MATCHES "^(armv7.*|armv8.*|aarch64.*)$")
  set(NEON_CPU ON)
else()
  set(NEON_CPU OFF)
endif()

option(NEON "Use ARM Neon SIMD instructions if in ARM CPU" ON)
if(${NEON} AND ${NEON_CPU})
  target_compile_definitions(SoundTouch PRIVATE SOUNDTOUCH_USE_NEON)
  if(NOT CMAKE_SYSTEM_PROCESSOR MATCHES "^aarch64.*$")
    target_compile_options(SoundTouch PRIVATE -mfpu=neon)
  endif()
endif()
```

So I guess CMAKE_SYSTEM_PROCESSOR is matching `armv7` or `armv8` on Buildbot? I thought Flathub builds for aarch64 though... :thinking:  @hfiguiere do you have any suggestions on this?

File: org.mixxx.Mixxx.yaml [hunk 12]
```diff
@@ -44,175 +56,291 @@ cleanup:
...
+      # Disable soundstretch utility and disable non-working -mfpu=neon flag
+      - -DSOUNDSTRETCH=OFF
+      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
+      - -DNEON=OFF
     sources:
       - type: archive
-        url: https://codeberg.org/soundtouch/soundtouch/archive/2.3.1.tar.gz
-        sha256: 42633774f372d8cb0a33333a0ea3b30f357c548626526ac9f6ce018c94042692
+        url: https://codeberg.org/soundtouch/soundtouch/archive/2.3.2.tar.gz
+        sha256: 714c8626001bb3ef291a9af9eadc0f7b382a0b23506efebc516b63b170b4da56
```
Comment (Be-ing) [inline org.mixxx.Mixxx.yaml:227]:
```suggestion
        sha256: ed714f84a3e748de87b24f385ec69d3bdc51ca47b7f4710d2048b84b2761e7ff
```

Codeberg changed the hash :/ Probably didn't break CI here yet due to caching.

File: org.mixxx.Mixxx.yaml [hunk 13]
```diff
@@ -44,175 +56,291 @@ cleanup:
...
+      - -Dladspa=disabled
+      - -Dlv2=disabled
     sources:
       - type: archive
-        url: https://github.com/breakfastquay/rubberband/archive/refs/tags/v2.0.2.tar.gz
-        sha256: 1372a0d52915ef07aa3cffc00a8b17996c736f328efa0f6504d231fa6428ed4b
+        url: https://github.com/breakfastquay/rubberband/archive/refs/tags/v3.3.0.tar.gz
+        sha256: 2bb837fe00932442ca90e185af8a468f7591df0c002b4a9e27a1bced1563ac84
 
+  # Audio processing Library
```
Comment (Be-ing) [inline org.mixxx.Mixxx.yaml:216]:
```suggestion
  # Audio time stretching and pitch shifting library
```

File: org.mixxx.Mixxx.yaml (hunk 5 above)
Comment (hfiguiere) [reply org.mixxx.Mixxx.yaml:38]: new that I'm reading the comment, persist here is for ~/.mixxx which is not what the comment says.

File: org.mixxx.Mixxx.metainfo.xml
Comment (hfiguiere) [inline org.mixxx.Mixxx.metainfo.xml:1]: why isn't this upstream?

File: org.mixxx.Mixxx.yaml [hunk 14]
```diff
@@ -44,175 +56,291 @@ cleanup:
...
+      url: https://github.com/mixxxdj/mixxx.git
+      commit: 5e00122dd5da4fa0d6f9a0f95291a6ea68743c80
+    - type: file
+      path: org.mixxx.Mixxx.metainfo.xml
     - type: script
       dest-filename: mixxx.sh
       commands:
-        - LV2_PATH=$HOME/.lv2:/app/extensions/Plugins/lv2:/app/lib/lv2
+        - export LV2_PATH=$HOME/.lv2:/app/extensions/Plugins/lv2:/app/lib/lv2
         - exec mixxx.bin $@
```
Comment (hfiguiere) [inline org.mixxx.Mixxx.yaml:346]:
```suggestion
        - exec mixxx.bin "$@"
```

File: org.mixxx.Mixxx.metainfo.xml
Comment (Be-ing) [reply org.mixxx.Mixxx.metainfo.xml:1]: Again, the upstream file is autogenerated.

File: org.mixxx.Mixxx.yaml (hunk 5 above)
Comment (hfiguiere) [reply org.mixxx.Mixxx.yaml:38]: no. because that's not what it does.

**Comment by Be-ing** [comment]
I updated libmad in the shared-modules repository (https://github.com/flathub/shared-modules/pull/292). Please update the git submodule in this PR.

**Comment by Be-ing** [comment]
Ping @FakeShemp this is ready to merge to update to the new upstream 2.4.0 release.
